docs/examples/plot_singlediode.py

```diff
@@ -0,0 +1,140 @@
+"""
+Single-diode equation
+=====================
+
+Examples of modeling IV curves using a single-diode circuit equivalent model.
+"""
+
+#%%
+# Calculating a module IV curve for certain operating conditions is a two-step
+# process.  Multiple methods exist for both parts of the process.  Here we use
+# the De Soto 5-parameter model to calculate the electrical characteristics
+# of a PV module at a certain irradiance and temperature using the module's
+# base characteristics at reference conditions.  Those parameters are then used
+# to calculate the module's IV curve by solving the single-diode equation using
+# the Lambert W method.
+#
+# The single-diode equation is a circuit-equivalent model of a PV
+# cell and has five electrical parameters that depend on the operating
+# conditions.  For more details on the single-diode equation and the five
+# parameters, see the `PVPMC single diode page
+# <https://pvpmc.sandia.gov/modeling-steps/2-dc-module-iv/diode-equivalent-circuit-models/>`_.
+#
+# Calculating IV Curves
+# -----------------------
+# This example uses :py:meth:`pvlib.pvsystem.calcparams_desoto` to calculate
+# the 5 electrical parameters needed to solve the single-diode equation.
+# :py:meth:`pvlib.pvsystem.singlediode` is then used to generate the IV curves.
+
+from pvlib import pvsystem
+import pandas as pd
+import matplotlib.pyplot as plt
+
+# Example module parameters for the Canadian Solar CS5P-220M:
+parameters = {
+    'Name': 'Canadian Solar CS5P-220M',
+    'BIPV': 'N',
+    'Date': '10/5/2009',
+    'T_NOCT': 42.4,
+    'A_c': 1.7,
+    'N_s': 96,
+    'I_sc_ref': 5.1,
+    'V_oc_ref': 59.4,
+    'I_mp_ref': 4.69,
+    'V_mp_ref': 46.9,
+    'alpha_sc': 0.004539,
+    'beta_oc': -0.22216,
+    'a_ref': 2.6373,
+    'I_L_ref': 5.114,
+    'I_o_ref': 8.196e-10,
+    'R_s': 1.065,
+    'R_sh_ref': 381.68,
+    'Adjust': 8.7,
+    'gamma_r': -0.476,
+    'Version': 'MM106',
+    'PTC': 200.1,
+    'Technology': 'Mono-c-Si',
+}
+
+base_case = {'Geff': 400, 'Tcell': 80}
```

Comment:
A suggestion to improve readability of the figure: order the cases from top to bottom, then right to left.
(1000, 80), (800, 80), (600, 80), (400, 30), (400, 55), (400, 80)

That way the legend order will correspond to the vertical order of the curves (top to bottom), and the direction of the Temperature arrow (right to left).

Entirely a quibble: 80C is pretty high for a cell temperature, perhaps 25, 40, 55 instead.

Reply:
I originally had labels in the plot itself and bumped up the temperature range a bit to get more space, but that's not needed since labels are in the legend now!  Example is updated.